helper/resource/teststep_providers_test.go

```diff
@@ -1738,43 +1923,41 @@ func composeImportStateCheck(fs ...ImportStateCheckFunc) ImportStateCheckFunc {
 	}
 }
 
-func testExtractResourceAttrInstanceState(attributeName string, attributeValue *string) ImportStateCheckFunc {
+func testExtractResourceAttrInstanceState(id, attributeName string, attributeValue *string) ImportStateCheckFunc {
```

Comment:
Nit: Instead of fetching via the internal identifier (typically the `id` attribute, but not necessarily guaranteed) can we instead use the resource address string? e.g. `random_string.example` Reasons being:

- This is how other testing framework "check" functions generally work
- Resources such as this one may have duplicate resource instances with the same id
- We're looking at making id less special in #1072

Reply:
Not sure how we use resource address when we're using a `type ImportStateCheckFunc func([]*terraform.InstanceState) error` as `*terraform.InstanceState` just has the `ID` and `Attributes` and doesn't appear to have any reference to the resource address.

Reply:
Sigh, more trouble than its worth then (e.g. mapping those instance states with the resource addresses). We'll be removing these `terraform` bits when we take a look at the next iteration of the test framework and this isn't exported, so fine for now.